control/injector/Injector.php

```diff
@@ -594,6 +594,20 @@ public function inject($object, $asType=null) {
 			}
 		}
 
+		// Populate named methods
+		if (isset($this->specs[$objtype]['methods'])) {
+			foreach ($this->specs[$objtype]['methods'] as $method) {
```

Comment:
This makes the assumption that `$this->specs[$objtype]['methods']` is an array. It might not be. Someone might have goofed something up.